### PR TITLE
BLD: since we already use setuptools, let's remove the optional logic…

### LINKED ISSUE #18113: BLD: since we already use setuptools, let's remove the optional logic in setup.py
no need to have the branching in setup.py we always have setuptools available.

### PR DESCRIPTION
**Pull request by gkonefal-reef**
… in setup.py (GH18113).

- [x] closes #18113 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -729,9 +709,7 @@ def pxd(name):
                       sources=['pandas/util/move.c'])
 extensions.append(_move_ext)
 
-
-if _have_setuptools:
-    setuptools_kwargs["test_suite"] = "nose.collector"
+setuptools_kwargs["test_suite"] = "nose.collector"
```
Comment (jreback) [inline setup.py]: remove this, its no longer needed

File: setup.py [hunk 2]
```diff
@@ -56,10 +50,6 @@ def is_platform_mac():
                                               'pytz >= 2011k',
```
Comment (jreback) [inline setup.py]: this can be a bit simpler, e.g. use a common setuptool_kwargs, leaving out datetutil, then add this separately for PY2 and 3

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.22.0.txt [hunk 3]
```diff
@@ -195,5 +195,6 @@ Other
 ^^^^^
 
 - Improved error message when attempting to use a Python keyword as an identifier in a numexpr query (:issue:`18221`)
+- BLD: since we already use setuptools, let's remove the optional logic in setup.py (:issue:`18113`)
```
Comment (bashtage) [inline doc/source/whatsnew/v0.22.0.txt]: This isn't the usual style of a what's new entry.  Something like "Simplified setup due to explicit dependence on setuptools (:issue...)"

**Comment by jreback** [comment]
can you rebase and update

**Comment by gkonefal-reef** [comment]
I've applied changes as it was requested.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ok re-running on travis [here](https://github.com/jreback/pandas/commit/c6b88e11268500c8cc26e397d66c30e0853d5022) just to make sure

**Comment by jreback** [comment]
thanks @gkonefal-reef

**Pull request closed without merging**